src/CoreMedia/CMSync.cs

```diff
@@ -699,6 +824,89 @@ public static bool MightDrift (CMClockOrTimebase clockOrTimebaseA, CMClockOrTime
 
 			return CMSyncMightDrift (clockOrTimebaseA.Handle, clockOrTimebaseB.Handle);
 		}
+
+#if !NET
+		[Watch (8,0), TV (15,0), Mac (12,0), iOS (15,0)]
+#else
+		[SupportedOSPlatform ("ios15.0"), SupportedOSPlatform ("tvos15.0"), SupportedOSPlatform ("macos12.0")]
+#endif
+		[DllImport (Constants.CoreMediaLibrary)]
+		static extern /* CMTimebase */ IntPtr CMTimebaseCopySourceTimebase (/* CMTimebase */ IntPtr timebase);
+
+#if !NET
+		[Watch (8,0), TV (15,0), Mac (12,0), iOS (15,0)]
+#else
+		[SupportedOSPlatform ("ios15.0"), SupportedOSPlatform ("tvos15.0"), SupportedOSPlatform ("macos12.0")]
+#endif
+		[DllImport(Constants.CoreMediaLibrary)]
+		static extern int CMTimebaseSetSourceTimebase (/* CMTimebase */ IntPtr timebase, /* CMTimebase */ IntPtr newSourceTimebase);
+
+#if !NET
+		[Watch (8,0), TV (15,0), Mac (12,0), iOS (15,0)]
+#else
+		[SupportedOSPlatform ("ios15.0"), SupportedOSPlatform ("tvos15.0"), SupportedOSPlatform ("macos12.0")]
+#endif
+		public CMTimebase? SourceTimebase  {
+			get {
+				var source = CMTimebaseCopySourceTimebase  (Handle);
+				return Runtime.GetINativeObject<CMTimebase> (source, true); 
+			}
+			set {
+				CMTimebaseSetSourceTimebase (Handle, value?.GetHandle () ?? IntPtr.Zero);
```

Comment:
The null check should be unnecessary:

```suggestion
				CMTimebaseSetSourceTimebase (Handle, value.GetHandle ());
```